File: smart-job-portal-frontend/src/services/jobService.js
import API from "../utils/axiosConfig";

export const getAllJobs = async () => {
    const response = await API.get("/jobs");
    return response.data;
};

export const searchJobs = async (keyword) => {
    const response = await API.get(`/jobs/search?keyword=${keyword}`);
    return response.data;
};
export const deleteJob = async (id) => {
    const response = await API.delete(`/jobs/${id}`);
    return response.data;
};

export const updateJob = async (id, jobData) => {
    const response = await API.put(`/jobs/${id}`, jobData);
    return response.data;
};

File: smart-job-portal-backend/src/main/java/com/jobportal/service/JobService.java
package com.jobportal.service;
import com.jobportal.dto.JobRequest;
import com.jobportal.dto.JobResponse;
import org.springframework.data.domain.Page;
public interface JobService {
    JobResponse createJob(JobRequest request);
    JobResponse updateJob(Long id, JobRequest request);
    void deleteJob(Long id);
    Page<JobResponse> getAllJobs(int page, int size);
    Page<JobResponse> searchJobs(String keyword, int page, int size);
}
